src/main/java/com/pwn9/filter/bukkit/listener/PwnFilterCommandListener.java

```diff
@@ -120,12 +120,9 @@ private void eventProcessor(PlayerCommandPreprocessEvent event) {
             // Simple Spam filter
             if (BukkitConfig.commandspamfilterEnabled() && !minecraftPlayer.hasPermission("pwnfilter.bypass.spam")) {
                 // Keep a log of the last message sent by this player.  If it's the same as the current message, cancel.
-                if (PwnFilterBukkitPlugin.lastMessage.containsKey(minecraftPlayer.getId()) && PwnFilterBukkitPlugin.lastMessage.get(minecraftPlayer.getId()).equals(message)) {
-                    event.setCancelled(true);
-                    minecraftPlayer.sendMessage(ChatColor.DARK_RED + "[PwnFilter]" + ChatColor.RED + " Repeated command blocked by spam filter.");
+                if(checkIfSpam(minecraftPlayer,message,event)){
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/gh/AddstarMC/PwnFilter/pullRequest?prid=6040651)